pykafka/balancedconsumer.py

```diff
@@ -741,10 +749,11 @@ def consumer_timed_out():
                 self._raise_worker_exceptions()
                 self._internal_consumer_running.wait(self._consumer_timeout_ms / 1000)
             try:
-                # acquire the lock to ensure that we don't start trying to consume from
-                # a _consumer that might soon be replaced by an in-progress rebalance
-                with self._rebalancing_lock:
-                    message = self._consumer.consume(block=block)
+                if block:
+                    while not message:
```

Comment:
I see what you're going for here, but if my read is correct, this amounts to busywaiting, which is something we want to avoid. This has been addressed in previous tickets https://github.com/Parsely/pykafka/issues/628, and I'd like to avoid reintroducing a condition under which it's possible to busywait in `consume()`.

Reply:
A way to fix this issue without incurring high-CPU waiting could be to have the balanced consumer adjust `self._consumer._messages_arrived`, since it's the semaphore that's [used](https://github.com/Parsely/pykafka/blob/5d222bdcbdad7b23e0d252f9b4f2eeb23dd89817/pykafka/simpleconsumer.py#L440) to control how the `consume()` call blocks. I believe if we were to release that semaphore like we do [here](https://github.com/Parsely/pykafka/blob/5d222bdcbdad7b23e0d252f9b4f2eeb23dd89817/pykafka/simpleconsumer.py#L952) and add some logic to `SimpleConsumer.consume()` to let it know that the semaphore was released without a message actually having arrived, we'd fix the same issue you're attempting to with this PR.
My only concern is over how to do this in a way that maintains the sanity of the interfaces through which these two consumer types communicate with each other. Adjusting private members for purposes other than their intended uses is a road I'd rather not go down.

Reply:
Right. One approach could be to have a timeout of 5s even for `block=True`
Anyways, this looks like a duplicate of #718